### DISCUSSION TITLE
How to look for overfitting?

### DISCUSSION THREAD

**Discussion by VDFaller** [open-edge-platform/anomalib#2566] [Q&A] [answered]
Is there a better way with this framework to look for overfitting?  Since it doesn't log train and val loss?  

Or is there a way to enable the logging of those without subclassing and ovewriting `training_step` and `validation_step`?

Just looking for best practices.

**Comment by alexriedel1** [comment, marked as answer]
You can log your training and validation results by using one of the available logging methods: https://anomalib.readthedocs.io/en/latest/markdown/guides/reference/loggers/index.html

**Reply by VDFaller** [reply]
Wait you pass what you want to log to your logger?  That's not how lightning works right? 
I'm already using the logger, but the losses aren't logged.  But like CFA only logs `image_AUROC` and `image_F1Score` (which seems like the default).  

Or FastFLow/Draem logs the train_loss, but not the val_loss.

**Reply by phillipkusinski** [reply]
Im currently working on a project where I am facing the same thoughts. I am training a EfficientAD model with my custom dataset and wanted to activate monitor="val_loss" in the ModelCheckpoint. After errors I encounterd, that val_loss is not available but only train_loss:

# define checkpoint_callback
checkpoint_callback = ModelCheckpoint(
    dirpath=f"checkpoints/{run_name}/",           
    filename=run_name + "_{epoch:02d}",
    save_last=False,                   
    save_top_k=1,                     
    monitor="train_loss",               
    mode="min",                       
)

Is val_loss just not supported yet or what is the best approach?

**Reply by alexriedel1** [reply]
> Im currently working on a project where I am facing the same thoughts. I am training a EfficientAD model with my custom dataset and wanted to activate monitor="val_loss" in the ModelCheckpoint. After errors I encounterd, that val_loss is not available but only train_loss:
> 
> # define checkpoint_callback
> checkpoint_callback = ModelCheckpoint( dirpath=f"checkpoints/{run_name}/", filename=run_name + "_{epoch:02d}", save_last=False, save_top_k=1, monitor="train_loss", mode="min", )
> 
> Is val_loss just not supported yet or what is the best approach?

EfficientAD does quite different things during training and testing so the validation loss is not really interesting and therefore is not supported. You could monitor the "image_f1score" for example or the "pixel_f1score" depending on what you are interested

**Reply by phillipkusinski** [reply]
Awesome thanks for the help!
